views/pulsar/v2/helpers/form.html.twig

```diff
@@ -2107,3 +2107,87 @@ data-*        | string | Data attributes, eg: `'data-foo': 'bar'`
+                    'label': options.label,
+                    'required': options.required|default(false)
+                })
+            }}
+            <div class="controls">
+                <div class="table-container">
+                    <table class="table table--full repeatable__table">
+                        <thead class="repeater__preview-headings">
+                            <tr>
+                                {% for heading in options.headings %}
+                                    <th data-repeater-for-name="{{ heading.name }}">{{ heading.label }}</th>
+                                {% endfor %}
+                                <th rowspan="1" colspan="1"></th>
+                            </tr>
+                        </thead>
+                        <tbody class="repeater__preview-data">
+                            <tr class="repeater__empty-placeholder">
+                                <td colspan="{{ headingLength }}" class="muted">{{ options.empty }}</td>
+                                <td class="table__td table__td--align-right table-row__actions"></td>
+                            </tr>
+                        </tbody>
+                        <tfoot>
+                        <tr class="repeater__group" data-repeater-new-group>
+                            <td class="table__form" colspan={{ headingLength + 1 }}>
+
+{% endspaceless %}
+{% endmacro %}
+
+{% macro repeatable_end() %}
+{% import '@pulsar/pulsar/v2/helpers/html.html.twig' as html %}
+{% spaceless %}
+                                <div class="form__actions repeater__group-actions">
+                                    {{
+                                        html.button({
+                                            'data-repeater-save-group': 'true',
+                                            'label': 'Save',
+                                            'class': 'btn--primary repeater__save-group',
+                                            'type': 'button'
+                                        })
+                                    }}
+
+                                    {{
+                                        html.button({
+                                            'data-repeater-cancel-save': 'true',
+                                            'label': 'Cancel',
+                                            'class': 'btn--naked repeater__cancel-group',
+                                            'type': 'button'
+                                        })
+                                    }}
+                                </div>
+                            </td>
+                        </tr>
+                        </tfoot>
+                    </table>
+                </div>
+                {{
+                    html.button({
+                        'data-repeater-add-group': 'true',
+                        'label': 'Add',
+                        'class': 'btn--primary repeater__add-group',
```

Comment:
shouldn't be a primary button

Reply:
What are your thoughts on what these actions should be in that case, I have a *Add* & *Save* (which is the same as update), both *cancel* actions seem to work with naked.

Reply:
You mean what style? Add & Save should just be regular grey buttons, the primary action would be higher up in the UI and be the main form save/submit action